lsp-completion.el

```diff
@@ -755,6 +755,20 @@ The CLEANUP-FN will be called to cleanup."
   "Disable LSP completion support."
   (lsp-completion-mode -1))
 
+(defun lsp-completion-passthrough-try-completion (string table pred point)
+  (let* ((completion-ignore-case t)
+         (try (completion-basic-try-completion string table pred point))
+         (newstr (car try))
+         (newpoint (cdr try))
+         (beforepoint (and try (substring newstr 0 newpoint))))
+    (if (and beforepoint
+             (string-prefix-p
+              beforepoint
+              (try-completion "" table pred)
+              t))
+        try
+      (cons string point))))
```

Comment:
Since this is a passthrough try completion, can we just return the `(cons string point)` instead of using `try-completion` here?

Reply:
It's possible - then "expand common" will always just keep the current input.

That seems like a bit of a loss from my POV, but VS Code never supported this anyway, so :shrug:

Reply:
Agree, the `basic-try-completion` is also matching prefix anyway, and in case of advanced language server that do the fuzzy match by themselves, I don't think we want to be smart and step in to decide where is the common prefix.

Reply:
This is my main feedback, for other comments, feel free to won't fix them.